plugins/catalog-backend/api-report.md

```diff
@@ -289,12 +297,32 @@ export const createCatalogPermissionRule: <TParams extends unknown[]>(
   rule: PermissionRule<Entity, EntitiesSearchFilter, 'catalog-entity', TParams>,
 ) => PermissionRule<Entity, EntitiesSearchFilter, 'catalog-entity', TParams>;
 
+// @alpha
+export function createCounterMetric<T extends string>(
+  config: CounterConfiguration<T>,
+): Counter<T>;
```

Comment:
Hm, should we export these right now or should we wait to make custom metrics into a separate thing? @benjdlambert One drawback here is that we start exporting `prom-client` types which is a potential risk for future compat problem when we want to upgrade that library to a newer version.

Reply:
I think we probably want to copy these for now to be honest. When the backend system work is getting shipped I'd expect that we have a `metricsApiRef` or something that everyone can use and we can provide a default implementation which uses `prom-client` under the hood.

Reply:
I also dislike the `scaffolder-backend` depending on the `catalog-backend` at all, we should move the things like move the `CatalogProcessor` etc imports to `catalog-node` instead.

Reply:
Oh wait, I'm blind - thanks for calling this out, I missed somehow that this was in catalog-backend and this is an absolute no-no and there is no reason to use any metrics export from the catalog. Building some facility into backend-common for declaring metrics isn't quite on the map right now, so the remaining option is indeed to copy this over into the scaffolder backend

Reply:
Thanks @freben and @benjdlambert I wasn't really sure what direction to suggest @spencerrichardhenry should go with this